lib/codegen/src/simple_preopt.rs

```diff
@@ -536,11 +542,279 @@ fn simplify(pos: &mut FuncCursor, inst: Inst) {
+            opcode: Opcode::Brif,
+            cond: br_cond,
+            destination,
+            ref args,
+        } => {
+            let first_arg = {
+                let args = pos.func.dfg.inst_args(inst);
+                args[0]
+            };
+            if let ValueDef::Result(iconst_inst, _) = pos.func.dfg.value_def(first_arg) {
+                if let InstructionData::BinaryImm {
+                    opcode: Opcode::IfcmpImm,
+                    imm: cmp_imm,
+                    arg: cmp_arg,
+                } = pos.func.dfg[iconst_inst]
+                {
+                    let cmp_imm: i64 = cmp_imm.into();
+                    if cmp_imm != 0 {
+                        return;
+                    }
+
+                    match br_cond {
+                        IntCC::NotEqual => BranchOptInfo {
+                            br_inst: inst,
+                            cmp_arg: cmp_arg,
+                            destination: destination,
+                            args: args.clone(),
+                            kind: BranchOptKind::NotEqualZero,
+                        },
+                        IntCC::Equal => BranchOptInfo {
+                            br_inst: inst,
+                            cmp_arg: cmp_arg,
+                            destination: destination,
+                            args: args.clone(),
+                            kind: BranchOptKind::EqualZero,
+                        },
+                        _ => return,
+                    }
+                } else {
+                    return;
+                }
+            } else {
+                return;
+            }
+        }
+        _ => return,
+    };
+
+    match info.kind {
+        BranchOptKind::EqualZero => {
+            let args = info.args.as_slice(&pos.func.dfg.value_lists)[1..].to_vec();
+            pos.func
+                .dfg
+                .replace(info.br_inst)
+                .brz(info.cmp_arg, info.destination, &args);
+        }
+        BranchOptKind::NotEqualZero => {
+            let args = info.args.as_slice(&pos.func.dfg.value_lists)[1..].to_vec();
+            pos.func
+                .dfg
+                .replace(info.br_inst)
+                .brnz(info.cmp_arg, info.destination, &args);
+        }
+    }
+}
+
+struct BranchOrderInfo {
+    term_inst: Inst,
+    term_inst_args: ValueList,
+    term_dest: Ebb,
+    cond_inst: Inst,
+    cond_arg: Value,
+    cond_inst_args: ValueList,
+    cond_dest: Ebb,
+    kind: BranchOrderKind,
+}
+
+enum BranchOrderKind {
+    BrzToBrnz,
+    BrnzToBrz,
+    InvertIntCond(IntCC),
+    InvertFloatCond(FloatCC),
+}
+
+fn branch_order(pos: &mut FuncCursor, cfg: &mut ControlFlowGraph, ebb: Ebb, inst: Inst) {
+    let info = match pos.func.dfg[inst] {
+        InstructionData::Jump {
+            opcode: Opcode::Jump,
+            destination,
+            ref args,
+        } => {
+            if let Some(next_ebb) = pos.func.layout.next_ebb(ebb) {
```

Comment:
Could you please do it like:

```rust
let next_ebb = if let Some(next_ebb) = pos.func.layout.next_ebb(ebb) {
    next_ebb
} else {
    return;
};

if destination == next_ebb {
    return;
}

// ...
```

Here and below to reduce the indentation significantly.